dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/task/DumpMetadataTask.java

```diff
@@ -24,21 +24,26 @@
 import java.io.Writer;
 import java.nio.charset.StandardCharsets;
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import org.anarres.jdiagnostics.ProductMetadata;
 
 /** @author shevek */
 public class DumpMetadataTask extends AbstractTask<Void>
     implements CoreMetadataDumpFormat.CompilerWorksDumpMetadataTaskFormat {
 
-  @Nonnull private final ConnectorArguments arguments;
+  @Nullable private final ConnectorArguments arguments;
   private final String format;
 
-  public DumpMetadataTask(@Nonnull ConnectorArguments arguments, @Nonnull String format) {
+  public DumpMetadataTask(@Nullable ConnectorArguments arguments, @Nonnull String format) {
```

Comment:
I'd recommend to keep the `Nonnull` constraint in the public constructor and add a private constructor without the constraint.

Optionally, composition or inheritance can be introduced to avoid special handling of null `arguments`.

Reply:
Done.